modules/market/src/main/java/com/opengamma/strata/market/curve/SeasonalNodalCurve.java

```diff
@@ -0,0 +1,638 @@
+        seasonalityDefinition.getAdjustmentType());
+  }
+
+  //-------------------------------------------------------------------------
+  // restricted constructor
+  @ImmutableConstructor
+  private SeasonalNodalCurve(
+      NodalCurve curve, 
+      DoubleArray seasonality, 
+      ShiftType adjustmentType) {
+    this.curve = curve;
+    this.seasonality = seasonality;
+    this.xFixing = curve.getXValues().get(0);
+    this.yFixing = curve.getYValues().get(0);
+    int i = seasonalityIndex(xFixing);
+    ArgChecker.isTrue(adjustmentType.applyShift(yFixing, seasonality.get(i) ) - yFixing < 1.0E-10, 
+        "Fixing value should be unadjusted");
+    this.adjustmentType = adjustmentType;
+  }
+
+  @Override
+  public CurveMetadata getMetadata() {
+    return curve.getMetadata();
+  }
+
+  @Override
+  public double yValue(double x) {
+    int i = seasonalityIndex(x);
+    double adjustment = seasonality.get(i);
+    return adjustmentType.applyShift(curve.yValue(x), adjustment);
+  }
+  
+  // The index on the seasonality vector associated to a time (nb months).
+  private int seasonalityIndex(double x) {
+    long xLong = Math.round(x);
+    return (int) ((xLong + 12) % 12); // Shift by 12 has java compute the remainder of negative numbers as negative
+  }
+
+  @Override
+  public int getParameterCount() {
+    return curve.getParameterCount() - 1;
+  }
+
+  @Override
+  public double getParameter(int parameterIndex) {
+    return curve.getParameter(parameterIndex + 1);
+  }
+
+  @Override
+  public ParameterMetadata getParameterMetadata(int parameterIndex) {
+    return curve.getParameterMetadata(parameterIndex + 1);
+  }
+
+  @Override
+  public DoubleArray getXValues() {
+    return curve.getXValues().subArray(1);
+  }
+
+  @Override
+  public DoubleArray getYValues() {
+    return curve.getYValues().subArray(1);
+  }
+
+  @Override
+  public UnitParameterSensitivity yValueParameterSensitivity(double x) {
+    int i = seasonalityIndex(x);
+    double adjustment = seasonality.get(i);
+    double derivativeFactor = 0.0;
+    if(adjustmentType.equals(ShiftType.ABSOLUTE)) {
+      derivativeFactor = 1.0;
+    } else if (adjustmentType.equals(ShiftType.SCALED)) {
+      derivativeFactor = adjustment;
+    } else {
+      throw new IllegalArgumentException("ShiftType " + adjustmentType + " is not supported for sensitivities");
+    }
+    UnitParameterSensitivity u = curve.yValueParameterSensitivity(x);
+    UnitParameterSensitivity u2 = 
+        UnitParameterSensitivity.of(u.getMarketDataName(), 
+            u.getParameterMetadata().subList(1, u.getParameterMetadata().size()), u.getSensitivity().subArray(1));
+    return u2.multipliedBy(derivativeFactor);
+  }
+
+  @Override
+  public double firstDerivative(double x) {
+    throw new UnsupportedOperationException("Value implemented only at discrete (monthly) values; no derivative available");
+  }
+
+  @Override
+  public NodalCurve withMetadata(CurveMetadata metadata) {
```

Comment:
Each of these `with` methods should return `SeasonalNodalCurve`, not `NodalCurve`.